tools/install.sh

```diff
@@ -54,49 +53,60 @@ main() {
   # precedence over umasks except for filesystems mounted with option "noacl".
   umask g-w,o-w
 
-  printf "${BLUE}Cloning Oh My Bash...${NORMAL}\n"
+  msg "${BLUE}Cloning Oh My Bash...${NORMAL}"
   hash git >/dev/null 2>&1 || {
-    echo "Error: git is not installed"
-    exit 1
+    die "git is not installed"
   }
   # The Windows (MSYS) Git is not compatible with normal use on cygwin
   if [ "$OSTYPE" = cygwin ]; then
     if git --version | grep msysgit > /dev/null; then
-      echo "Error: Windows/MSYS Git is not supported on Cygwin"
-      echo "Error: Make sure the Cygwin git package is installed and is first on the path"
-      exit 1
+      err "Windowr/MSYS Git is not supported on Cygwin"
+      die "Error: Make sure the Cygwin git package is installed and is first on the path"
     fi
   fi
-  env git clone --depth=1 https://github.com/ohmybash/oh-my-bash.git $OSH || {
-    printf "Error: git clone of oh-my-bash repo failed\n"
-    exit 1
-  }
+  env git clone --depth=1 https://github.com/ohmybash/oh-my-bash.git "${OSH}" \
+    || die "git clone of oh-my-bash repo failed"
 
-  printf "${BLUE}Looking for an existing bash config...${NORMAL}\n"
-  if [ -f $HOME/.bashrc ] || [ -h $HOME/.bashrc ]; then
-    printf "${YELLOW}Found ~/.bashrc.${NORMAL} ${GREEN}Backing up to ~/.bashrc.pre-oh-my-bash${NORMAL}\n";
-    mv $HOME/.bashrc $HOME/.bashrc.pre-oh-my-bash;
+  msg "${BLUE}Looking for an existing bash config...${NORMAL}\n"
+  if [ -f "${HOME}/.bashrc" ] || [ -h "${HOME}/.bashrc" ]; then
+    msg "${YELLOW}Found ~/.bashrc.${NORMAL}"
+    msg "${GREEN}Backing up to ~/.bashrc.pre-oh-my-bash${NORMAL}"
+    mv "${HOME}/.bashrc" "${HOME}/.bashrc.pre-oh-my-bash"
   fi
 
-  printf "${BLUE}Using the Oh My Bash template file and adding it to ~/.bashrc${NORMAL}\n"
-  cp $OSH/templates/bashrc.osh-template $HOME/.bashrc
-  sed "/^export OSH=/ c\\
-export OSH=$OSH
-  " $HOME/.bashrc > $HOME/.bashrc-ombtemp
-  mv -f $HOME/.bashrc-ombtemp $HOME/.bashrc
+  msg "${BLUE}Using the Oh My Bash template file and adding it to ~/.bashrc${NORMAL}"
+  cp "${OSH}/templates/bashrc.osh-template" "${HOME}/.bashrc"
+
+  # update bashrc template to point to osh
+  sed -i "s/^export OSH=.*$/export OSH=${OSH}" "${HOME}/.bashrc"
```

Comment:
`sed -i` is a GNU extension. Not every `sed` implementations support `-i`. Even for the `sed` implementations that support the option `-i`, there is variations in the behavior when `-i` is used without the option argument.

Reply:
I did not know that. Even so there should be a better way to update the template than moving it, passing it through sed to a new file and then overwriting the original.

Reply:
Yeah, actually the first `cp` is unneeded. We can directly feed `sed` with `bashrc.osh-template`.

On the other hand, I believe we should finally replace `.bashrc` using `mv` as in the current master. If we directly rewrite the file contents using `sed` or `cp`, when the rewriting failed unexpectedly by some other factors, there remains an incomplete `.bashrc` file. We should first create a complete file and then, only after it succeeded, replace the old `.bashrc` with the new complete `.bashrc`. The filesystem operation of `mv` is atomic so there is a guarantee that it will not result in a broken status even on the filesystem failure.